ktor-io/posix/src/io/ktor/utils/io/core/BufferFactoryNative.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.utils.io.core
+
+import io.ktor.utils.io.core.internal.*
+import io.ktor.utils.io.pool.*
+import kotlin.native.concurrent.ThreadLocal
+
+internal actual val DefaultChunkedBufferPool: ObjectPool<ChunkBuffer>
+    get() = pool!!
+
+/**
+ * Initialization hack because kotlin-native init order.
```

Comment:
`ThreadLocal` is initialized after all other top-level variables, so access to this variable from the other top-scope object leads to `BAD_ACCESS` in runtime because of `nullptr`

Reply:
Can you add this to a comment?

Reply:
Sure